### PR TITLE
add autentication logic

### PR DESCRIPTION
**Pull request by Sereg20**
Implement Authentication logic except error case

### CODE REVIEW

File: uimodule/webapp/controller/ListReport.controller.js [hunk 1]
```diff
@@ -35,5 +35,46 @@ sap.ui.define([
                 });
             }
         },
+
+        onSignInPress() {
+            const oActiveUser = this.getView().getModel().getProperty("/ActiveUser");
```
Comment (Artemische) [inline uimodule/webapp/controller/ListReport.controller.js:40]: this.getModel()

File: uimodule/webapp/Firebase.js [hunk 2]
```diff
@@ -31,15 +30,25 @@ sap.ui.define([
...
                 Questions: []
-            });            
+            });
 		},
 
         handleOnMethodSuccess(snapshotData) {
-            var fbModel = sap.ui.getCore().byId("container-ohranaTryda---idAppControl").getModel("fbModel");
+            var fbModel = sap.ui.getCore().byId("container-ohranaTryda---idAppControl").getModel();
             fbModel.setProperty("/Users", snapshotData);
         },
```
Comment (Artemische) [inline uimodule/webapp/Firebase.js:53]: const

File: uimodule/webapp/Firebase.js [hunk 3]
```diff
@@ -20,7 +20,6 @@ sap.ui.define([
 	return {
 
 		initializeFirebase() {
-			// Initialize Firebase with the Firebase-config
 			firebase.initializeApp(firebaseConfig);
             var db = firebase.database();
             var usersRef = db.ref("Users/");
```
Comment (Artemische) [inline uimodule/webapp/Firebase.js:25]: const

File: uimodule/webapp/Firebase.js [hunk 4]
```diff
@@ -31,15 +30,25 @@ sap.ui.define([
                 console.log("Error: " + error.code);
             });
             
-			// Return JSON Model
             return new JSONModel({
```
Comment (Artemische) [inline uimodule/webapp/Firebase.js:33]: mb move model structure to model file
like new JModel(model.createModel())

File: uimodule/webapp/controller/ListReport.controller.js [hunk 5]
```diff
@@ -35,5 +35,46 @@ sap.ui.define([
...
+                oLocalModel.setProperty("/ActiveUser", oExistingUser);
+                this._setAvailableUsers();
+            } else {
+                debugger
+                // handle user doesn't exist
+            }
+        },
+
+        _setAvailableUsers() {
+            const oLocalModel = this.getView().getModel();
```
Comment (Artemische) [inline uimodule/webapp/controller/ListReport.controller.js:62]: this.getModel()

File: uimodule/webapp/controller/ListReport.controller.js [hunk 6]
```diff
@@ -35,5 +35,46 @@ sap.ui.define([
...
+            const oLoginCredentials = {
+                name: oActiveUser.name,
+                secondName: oActiveUser.secondName,
+                thirdName: oActiveUser.thirdName
+            };
+            this._authenticateUser(oLoginCredentials);
+        },
+
+        _authenticateUser(oLoginCredentials) {
+            const oLocalModel = this.getView().getModel();
```
Comment (Artemische) [inline uimodule/webapp/controller/ListReport.controller.js:50]: this.getModel()

File: uimodule/webapp/controller/ListReport.controller.js [hunk 7]
```diff
@@ -35,5 +35,46 @@ sap.ui.define([
...
+            } else {
+                debugger
+                // handle user doesn't exist
+            }
+        },
+
+        _setAvailableUsers() {
+            const oLocalModel = this.getView().getModel();
+            const bActiveUserIsAdmin = oLocalModel.getProperty("/ActiveUser/isAdmin");
+            if (bActiveUserIsAdmin) {
```
Comment (Artemische) [inline uimodule/webapp/controller/ListReport.controller.js:64]: const sProperty = bActiveUserIsAdmin ? "/Users" : "/ActiveUser";
oLocalModel.setProperty(..., sProperty)

File: uimodule/webapp/controller/ListReport.controller.js [hunk 8]
```diff
@@ -35,5 +35,46 @@ sap.ui.define([
...
+                oLocalModel.setProperty("/AvailableUsers", oLocalModel.getProperty("/Users"));
+            } else {
+                oLocalModel.setProperty("/AvailableUsers", oLocalModel.getProperty("/ActiveUser"));
+            }
+        },
+
+        _verifyLoginUserExist(oLoginUser) {
+            const oFbWsResponseModel = this.getView().getModel();
```
Comment (Artemische) [inline uimodule/webapp/controller/ListReport.controller.js:72]: this.getModel

File: uimodule/webapp/controller/ListReport.controller.js (hunk 6 above)
Comment (Artemische) [reply uimodule/webapp/controller/ListReport.controller.js:50]: mb change variable name to 'oModel' (we dont have variable oRemoteModel)? 
```suggestion
            const oModel = this.getView().getModel();
```

**Review by Artemische — CHANGES_REQUESTED** [review]

File: uimodule/webapp/controller/ListReport.controller.js (hunk 5 above)
Comment (Sereg20) [reply uimodule/webapp/controller/ListReport.controller.js:62]: Fixed

File: uimodule/webapp/Firebase.js (hunk 3 above)
Comment (Sereg20) [reply uimodule/webapp/Firebase.js:25]: Fixed

File: uimodule/webapp/Firebase.js (hunk 2 above)
Comment (Sereg20) [reply uimodule/webapp/Firebase.js:53]: Fixed

File: uimodule/webapp/controller/ListReport.controller.js (hunk 1 above)
Comment (Sereg20) [reply uimodule/webapp/controller/ListReport.controller.js:40]: Fixed

File: uimodule/webapp/controller/ListReport.controller.js (hunk 8 above)
Comment (Sereg20) [reply uimodule/webapp/controller/ListReport.controller.js:72]: Fixed

File: uimodule/webapp/Firebase.js (hunk 4 above)
Comment (Sereg20) [reply uimodule/webapp/Firebase.js:33]: Fixed

**Pull request merged**